src/server/mod.rs

```diff
@@ -1474,26 +1474,25 @@ impl LanguageServer for LspServer {
                     .composition_string()
                     .expect("bad composition state");
 
-                // index 0 is line 1, char 1 for line_col::LineColLookup.
-                // Will need to convert to zero indexing for applyEdit. (Range is zero indexed.)
-                // https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#range
                 let last_pos = match old_text {
-                    Some(text) => line_col::LineColLookup::new(&text)
-                        .get(text.len()),
-                    None => line_col::LineColLookup::new(&new_text)
-                        .get(new_text.len()),
+                    Some((_, range_pos)) => range_pos,
+                    // applyEdit insertion point should be at Postion.start for new_text
+                    None => (new_text.1 .0, new_text.1 .1),
```

Comment:
This is the initialize() method, which is the only place where we may not have `old_text`.
In that case --> the `new_text` has the position we should use.

Reply:
Once again, if you feel this is important enough to put in the PR, that might indicate it's important enough to keep in the source. PRs are not documentation.